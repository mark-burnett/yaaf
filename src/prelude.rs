pub use crate::{
    actor::{Actor, Tell},
    context::{Context, Publish},
    handler::Handler,
    source::Source,
    system::System,
};
pub use ::async_trait::async_trait;
pub use ::yaaf_macros::{Actor, Source};
