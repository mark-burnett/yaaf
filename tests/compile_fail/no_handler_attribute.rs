use ::yaaf::prelude::*;

#[derive(Clone, Debug)]
struct MyMessage;

#[derive(Actor)]
struct MyActor;

#[async_trait]
impl Handler<MyMessage> for MyActor {
    async fn handle(&mut self, _ctx: &mut HandleContext, _message: MyMessage) {
    }
}

fn main() {}
