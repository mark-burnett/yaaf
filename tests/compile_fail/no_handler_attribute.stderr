error[E0277]: the trait bound `MyActor: yaaf::handler::detail::HandlesList<()>` is not satisfied
 --> $DIR/no_handler_attribute.rs:6:10
  |
6 | #[derive(Actor)]
  |          ^^^^^ the trait `yaaf::handler::detail::HandlesList<()>` is not implemented for `MyActor`
  |
 ::: $WORKSPACE/src/actor.rs
  |
  | pub trait Actor: HandlesList<<Self as Actor>::Handles> {
  |                  ------------------------------------- required by this bound in `yaaf::Actor`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
